src/Illuminate/Support/HigherOrderCollectionProxy.php

```diff
@@ -54,7 +54,15 @@ public function __get($key)
     public function __call($method, $parameters)
     {
         return $this->collection->{$this->method}(function ($value) use ($method, $parameters) {
-            return $value->{$method}(...$parameters);
+            if (is_object($value)) {
+                return $value->{$method}(...$parameters);
+            }
+
+            if ($method == 'empty') {
```

Comment:
Maybe I'm overlooking this, but the `empty` function isn't defined anyway?

https://3v4l.org/QZ1Rk